includes/classes/InternalConnections/NetworkSiteConnection.php

```diff
@@ -278,41 +288,22 @@ public function pull( $items ) {
 				}
 			}
 
-			foreach ( $post_props as $key => $value ) {
-				$post_array[ $key ] = $value;
-			}
-
-			if ( ! empty( $item_array['post_id'] ) ) {
-				$post_array['ID'] = $item_array['post_id'];
-			} else {
-				unset( $post_array['ID'] );
-			}
-
-			if ( isset( $post_array['post_parent'] ) ) {
-				unset( $post_array['post_parent'] );
-			}
-
-			if ( ! empty( $item_array['post_status'] ) ) {
-				$post_array['post_status'] = $item_array['post_status'];
-			}
-
 			add_filter( 'wp_insert_post_data', array( '\Distributor\InternalConnections\NetworkSiteConnection', 'maybe_set_modified_date' ), 10, 2 );
 
 			// Filter documented in includes/classes/ExternalConnections/WordPressExternalConnection.php
 			$new_post_args = Utils\post_args_allow_list( apply_filters( 'dt_pull_post_args', $post_array, $item_array['remote_post_id'], $post, $this ) );
-			$new_post_id   = wp_insert_post( wp_slash( $new_post_args ) );
-
+			if ( $update ) {
+				$new_post_id = wp_update_post( wp_slash( $new_post_args ) );
+			} else {
+				$new_post_id = wp_insert_post( wp_slash( $new_post_args ) );
+			}
 			remove_filter( 'wp_insert_post_data', array( '\Distributor\InternalConnections\NetworkSiteConnection', 'maybe_set_modified_date' ), 10, 2 );
 
 			if ( ! is_wp_error( $new_post_id ) ) {
-				update_post_meta( $new_post_id, 'dt_original_post_id', (int) $item_array['remote_post_id'] );
 				update_post_meta( $new_post_id, 'dt_original_blog_id', (int) $this->site->blog_id );
-				update_post_meta( $new_post_id, 'dt_syndicate_time', time() );
-				update_post_meta( $new_post_id, 'dt_original_post_url', esc_url_raw( $post->link ) );
-
-				if ( ! empty( $post->post_parent ) ) {
-					update_post_meta( $new_post_id, 'dt_original_post_parent', (int) $post->post_parent );
-				}
+				update_post_meta( $new_post_id, 'dt_syndicate_time', (int) time() );
```

Comment:
@peterwilsoncc time and post id can not be negative. If we convert them to integers, can we use `absint`?

Reply:
Resolved in 27b729cb54299bb79cf0b9ab192631d696070f0d